### DISCUSSION TITLE
Is it possible to skip the Parser?

### DISCUSSION THREAD

**Discussion by xiaoxinghu** [unifiedjs/unified#104] [Q&A] [answered]
If I already have a syntax tree at hand. I want to use unified to build a processor to process some of the nodes separately. So instead of having the text as input, the input is an AST. When I do that I got an error says that I don't have a parser in my processor. Currently, I call functions (e.g. `mdast-util-to-hast` etc.) directly and pass the AST in, one after another. It's okay, but it'd be nice to be able to leverage unified and it's plugins, like `rehype-highlight`.

**Comment by wooorm** [comment, marked as answer]
See https://github.com/unifiedjs/unified#description. unified has separate functions for parse/run/stringify

**Reply by xiaoxinghu** [reply]
OHHH! Awesome! Don't know how did I miss that. Thanks a lot! 👍
